### PR TITLE
take into account of vhd distro type for containerruntime provisioning [3/4]

### PR DESCRIPTION
**Pull request by eriksywu**
should not merge until https://github.com/Azure/AgentBaker/pull/361 is merged

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -100,17 +100,49 @@ const (
...
 const (
-	Ubuntu               Distro = "ubuntu"
-	Ubuntu1804           Distro = "ubuntu-18.04"
-	Ubuntu1804Gen2       Distro = "ubuntu-18.04-gen2"
-	AKSUbuntu1604        Distro = "aks-ubuntu-16.04"
-	AKSUbuntu1804        Distro = "aks-ubuntu-18.04"
-	AKSUbuntuGPU1804     Distro = "aks-ubuntu-gpu-18.04"
-	AKSUbuntuGPU1804Gen2 Distro = "aks-ubuntu-gpu-18.04-gen2"
+	Ubuntu                         Distro = "ubuntu"
+	Ubuntu1804                     Distro = "ubuntu-18.04"
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:118]: not sure if we need these two distros

File: pkg/agent/datamodel/types.go [hunk 2]
```diff
@@ -100,17 +100,49 @@ const (
...
+	AKSUbuntu1804                  Distro = "aks-ubuntu-18.04"
+	AKSUbuntuGPU1804               Distro = "aks-ubuntu-gpu-18.04"
+	AKSUbuntuGPU1804Gen2           Distro = "aks-ubuntu-gpu-18.04-gen2"
+	AKSUbuntuContainerd1804        Distro = "aks-ubuntu-containerd-18.04"
+	AKSUbuntuContainerd1804Gen2    Distro = "aks-ubuntu-containerd-18.04-gen2"
+	AKSUbuntuGPUContainerd1804     Distro = "aks-ubuntu-gpu-containerd-18.04"
+	AKSUbuntuGPUContainerd1804Gen2 Distro = "aks-ubuntu-gpu-containerd-18.04-gen2"
 )
 
+var ubuntu1804Distros = map[Distro]struct{}{
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:130]: ubuntu-18.04-gen2

**Comment by eriksywu** [comment]
Perhaps this change isn't needed if we enforce the correct containerruntime field on rp side

**Comment by eriksywu** [comment]
not needed - implemented/enforced on rp side. abandoning.

**Pull request closed without merging**